src/geometry/LinearPropagator.hh

```diff
@@ -20,19 +20,22 @@ class LinearPropagator
 {
     using Initializer_t = GeoStateInitializer;
 
+    //! Output results
+    struct result_type
+    {
+        real_type distance; //!< Distance traveled
+        VolumeId  volume;   //!< Post-propagation volume
+    };
+
   public:
-    // Construct from persistent and state data
+    //! Construct from persistent and state data
```

Comment:
Inline doxygen documentation like this should only be used if the function is inlined into the class declaration. Look at the following example for documenting inlined functions and inline function with out-of-line definitions.
```c++
struct Foo
{
    //! Do a thing
    void do_it() { this->do_it_impl(); }
};

struct Bar
{
    // Do a thing
    inline void do_it();
};

//---------------------------------------------------------------------------//
/*!
 * Do a thing.
 *
 * This does some sort of thing, which does things.
 */
void Bar::do_it()
{
    // whatever
}
```